Documentation/Evolution/UnicodeForStringProcessing.md

```diff
@@ -0,0 +1,691 @@
+let regex4 = Regex {
+    "ba"
+    "na".ignoringCase(false)
+    "na"
+}
+.ignoringCase()
+	
+"banana".contains(regex4)     // true
+"BAnaNA".contains(regex4)     // true
+"BANANA".contains(regex4)     // false
+
+// Equivalent to:
+let regex5 = /(?i)ba(?-i:na)na/
+```
+
+#### Case insensitivity
+
+Regular expressions perform case sensitive comparisons by default. The `i` option or the `ignoringCase(_:)` method enables case insensitive comparison.
+
+```swift
+let str = "Café"
+	
+str.firstMatch(of: /CAFÉ/)          // nil
+str.firstMatch(of: /(?i)CAFÉ/)      // "Café"
+str.firstMatch(of: /(?i)cAfÉ/)      // "Café"
+```
+
+Case insensitive matching uses case folding to ensure that canonical equivalence continues to operate as expected.
+
+**Regular expression syntax:** `(?i)...` or `(?i:...)`
+
+**`RegexBuilder` API:**
+
+```swift
+extension RegexComponent {
+    /// Returns a regular expression that ignores casing when matching.
+    public func ignoringCase(_ ignoreCase: Bool = true) -> Regex<Output>
+}
+```
+
+#### Single line mode (`.` matches newlines)
+
+The "any" metacharacter (`.`) matches any character in a string *except* newlines by default. With the `s` option enabled, `.` matches any character including newlines.
+
+```swift
+let str = """
+    <<This string
+    uses double-angle-brackets
+    to group text.>>
+    """
+    
+str.firstMatch(of: /<<.+>>/)        // nil
+str.firstMatch(of: /(?s)<<.+>>/)    // "This string\nuses double-angle-brackets\nto group text."
+```
+
+This option also affects the behavior of `CharacterClass.any`, which is designed to match the behavior of the `.` regex literal component.
+
+**Regular expression syntax:** `(?s)...` or `(?s...)`
+
+**`RegexBuilder` API:**
+
+```swift
+extension RegexComponent {
+  /// Returns a regular expression where the start and end of input
+  /// anchors (`^` and `$`) also match against the start and end of a line.
+  public func dotMatchesNewlines(_ dotMatchesNewlines: Bool = true) -> Regex<Output>
+}
+```
+
+#### Reluctant quantification by default
+
+Regular expression quantifiers (`+`, `*`, and `?`) match eagerly by default, such that they match the longest possible substring. Appending `?` to a quantifier makes it reluctant, instead, so that it matches the shortest possible substring.
+
+```swift
+let str = "<token>A value.</token>"
+	
+// By default, the '+' quantifier is eager, and consumes as much as possible.
+str.firstMatch(of: /<.+>/)          // "<token>A value.</token>"
+	
+// Adding '?' makes the '+' quantifier reluctant, so that it consumes as little as possible.
+str.firstMatch(of: /<.+?>/)         // "<token>"
+```
+
+The `U` option toggles the "eagerness" of quanitifiers, so that quantifiers are reluctant by default, and only become eager when `?` is added to the quantifier.
+
+```swift
+// '(?U)' toggles the eagerness of quantifiers:
+str.firstMatch(of: /(?U)<.+>/)      // "<token>"
+str.firstMatch(of: /(?U)<.+?>/)     // "<token>A value.</token>"
+```
+
```

Comment:
Should we consider a "possessive-by-default", either now or as future API?

Reply:
We could — that would probably necessitate figuring out some syntactical way of writing it, which could be a bit tricky. Maybe when possessive is the default then `.?+` is greedy? I can add it as another example to the future option directions.